### ISSUE TITLE
AverageFrames returns garbage from f32 input

### ISSUE THREAD

**Issue by NSQY** [vapoursynth/vapoursynth#869]
https://slow.pics/c/SvKc7DoP

R58 git master
Linux 5.17.3

```
import vapoursynth as vs
core = vs.core

clip = core.std.BlankClip(None, width=1920, height=1080, format=vs.RGBS, color=[.1]*3)
avg = core.std.AverageFrames(clip, weights=[1]*3, planes=[0, 1, 2])

avg.set_output()
```

**Comment by theChaosCoder** [comment]
Same problem on R57. Tested with: Size: 720x480 | Format: YUV420PS

**Comment by sekrit-twc** [comment]
bb10e48 workaround is to manually set the scale argument

**Comment by NSQY** [comment]
I regret to report that this is still broken, the horizontal streaks in the slow.pics example (https://slow.pics/c/5erFrUud) are still present.

Example: https://slow.pics/c/IBl9TH0J

```
import vapoursynth as vs
core = vs.core

blank = core.std.BlankClip(format=vs.YUV444PS, color=[0.5, 0, 0])
grain = core.grain.Add(blank, 50, 50, seed=422)
average = core.std.AverageFrames(grain, weights=[1] * 3)

blank.set_output(1)
average.set_output(2)
```

**Comment by sekrit-twc** [comment]
Fixed for real in da7d758

**Comment by NSQY** [comment]
Thank you!
